### ISSUE TITLE
Change the size of the DosBox-X

### ISSUE THREAD

**Issue by Arthur3032** [joncampbell123/dosbox-x#4316]
### What you want

Drag the window to change the size of the DosBox-X window
like other windows

### Have you checked that no similar feature request(s) exist?

- [X] I have searched and didn't find any similar feature request.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by joncampbell123** [comment]
What do you mean? DOSBox-X has supported window resize for years now.

The only restriction is that for output=surface you cannot resize below the size of the screen.

**Comment by Arthur3032** [comment]
Like This, when output=ttf

https://github.com/joncampbell123/dosbox-x/assets/93595854/7601d5bf-f258-4477-bbfe-279cf56244cc

**Comment by Arthur3032** [comment]
How to set -cols and -lins in command line?

**Comment by Arthur3032** [comment]
How to use commands to change the window resolution while using DoSbox-X.
ex:
resolu 100 30
resolu 80 25
Just like using TITLE to change the window title

**Comment by Arthur3032** [comment]
Thank you!
This is a good way. 
But after the resolution is changed, the data display position is wrong.

**Comment by Arthur3032** [comment]
[TRESU.zip](https://github.com/joncampbell123/dosbox-x/files/11811585/TRESU.zip)

**Comment by Arthur3032** [comment]
this is 80x25:
![image](https://github.com/joncampbell123/dosbox-x/assets/93595854/27549f7a-10fa-4fb9-8a38-4c5f7b4cc9a1)

press [F2] change to 132x43, shown as below:
![image](https://github.com/joncampbell123/dosbox-x/assets/93595854/360a5575-c724-462e-b925-e4dba1e1de2b)

**Comment by rderooy** [comment]
The DOS program must support text modes other than the common 80x25. If it does not, you get such results.

**Comment by Arthur3032** [comment]
> The DOS program must support text modes other than the common 80x25. If it does not, you get such results.

80x25 & 132x43 have support,
But the runtime changes it, and it doesn't get displayed normally

**Comment by Arthur3032** [comment]
How to switch display mode between 80x25 and 100x30
When executing my program in Dosbox-X

**Comment by maron2000** [comment]
I don't understand why you are asking this since you are already aware of the existence of `132X25.com`.
Anyway you can run `132x43.com` in advance of running your program.
And you need to check that your program adequately detects the change of the number of rows and lines, and redraw the screen accordingly.

![tresu](https://github.com/joncampbell123/dosbox-x/assets/68574602/a8aef4f6-c78e-4bfa-a414-30dcddac92a0)

**Comment by Arthur3032** [comment]
After 132X25 becomes 132x43
I have to redraw the screen.
However, the position displayed by 132x43 is not normal.

If, CONF is set to 132X43, enter DOSBOX-X, the display is normal.
CONF is set to 132X25, enter DOSBOX-X, the display is also normal

If the resolution is changed during RUNTIME, the display will be abnormal after the change

**Comment by maron2000** [comment]
Yes, so that means that your program isn't detecting the change of screen.
On the left-upper corner of your screenshot, it shows (24,79) regardless of the actual screen, which is the reason of the glitch.

**Comment by Arthur3032** [comment]
[RESOLU.zip](https://github.com/joncampbell123/dosbox-x/files/11843584/RESOLU.zip)

execute RESOLU.exe to test change resolution.
choice 1,2,3 is OK
4,5,6,7,8 is not OK
in RESOLU.exe is to RUN 80x25.com / 80x43.com......

**Comment by Arthur3032** [comment]
> Yes, so that means that your program isn't detecting the change of screen.
> On the left-upper corner of your screenshot, it shows (24,79) regardless of the actual screen, which is the reason of the glitch.

but, I set the resolution in doxbox-x.conf, and then execute my program , it can find maxrow(), maxcol().

So after executing other resolutions, dosbox-x should also execute something, making it like when dosbox-x was just started

**Comment by maron2000** [comment]
Your program initially detects maxrows and columns, but after it is changed, your have to tell your program to detect it again, or force overwrite maxrows and maxcols if possible.
If your program is not capable of doing so, there is now way that DOSBox-X will do that for you so you have to give it up.
Therefore, further discussions are not apporopriate, and you should conault a forum of the programming language you are using.

**Comment by Arthur3032** [comment]
> Your program initially detects maxrows and columns, but after it is changed, you have to tell your program to detect it again, or force overwrite maxrows and maxcols if possible. If your program is not capable of doing so, there is no way that DOSBox-X will do that for you so you have to give it up. Therefore, further discussions are not apporopriate, and you should consult a forum of the programming language you are using.
> 
> Edit: Fixed typos, sorry for that.

After I change the resolution, there is a state of re-detecting the resolution,
So 80x25, 80x43, 80x50 will be correct.
And 132x43... can't be detected,
But when dosbox-x is first started, they are all detectable
